### DISCUSSION TITLE
Import Suunto fit files battery_level

### DISCUSSION THREAD

**Discussion by mff73** [mytourbook/mytourbook#1284] [Ideas]
I just discover recently that suunto files files are now containing new informations about battery_level during activities.
In fact, it seems it appears around Oct 10th 2023.
battery_level is in device_info message

There seems to be some fields in MTB for battery levels, but i don't think suunto's one are imported, or at least, i didn't see anything.

**Comment by FJBDev** [comment]
> There seems to be some fields in MTB for battery levels, but i don't think suunto's one are imported, or at least, i didn't see anything.

If the Suunto FIT file contains the battery information, it should be imported just like any other FIT file from Garmin etc...
The import of [battery info](https://github.com/mytourbook/mytourbook/issues/376) was added in [MT 21.12](https://mytourbook.sourceforge.io/mytourbook/index.php/history/history-2021#v21.12.0)

![image](https://github.com/mytourbook/mytourbook/assets/25235665/ea1b5c72-6a59-4a0c-bc03-b72a6625f40c)

**Reply by mff73** [reply]
Thanks, but i am not speaking about a sensor battery, but watch battery level.
For example this file :
[suunto file with battery_level.zip](https://github.com/mytourbook/mytourbook/files/14175429/suunto.file.with.battery_level.zip)
is containing these data
![image](https://github.com/mytourbook/mytourbook/assets/50132146/81ee9c05-c7fd-4397-bb22-d978bc95fc53)

is MTB currently importing such data ? 🤔

**Reply by FJBDev** [reply]
Nevermind, I mixed it with the sensor battery level.
I know the data is not new because it was in the former Suunto XML/SML format as [here](https://raw.githubusercontent.com/mytourbook/mytourbook/main/bundles/net.tourbook.tests/src/test/device/suunto/files/log-F783095113000500-2013-05-18T11_00_38-0.xml)

```
<BatteryChargeAtStart>0.63</BatteryChargeAtStart>
<BatteryCharge>0.37</BatteryCharge>
```

**Comment by wolfgang-ch** [comment]
Currently the battery data are imported with an undocumented API because when it was implemented, there was no offical battery API. This API is now available and should work with the Suuonto device.

**Comment by mff73** [comment]
Sorry i am not technical enough to understand : 
you are confirming these battery data (at least the ones from Suunto) are not yet imported in MTB, but it could be in the future, if a proper change is done in MTB, right ?
Or you are saying that it should already work ?
🤔

**Comment by wolfgang-ch** [comment]
The battery data are available in the fit file and can be read but MT is not yet supporting the import of these battery data

**Comment by wolfgang-ch** [comment]
I created a feature request https://github.com/mytourbook/mytourbook/issues/1287

**Comment by wolfgang-ch** [comment]
> Thanks, but i am not speaking about a sensor battery, but watch battery level. For example this file : [suunto file with battery_level.zip](https://github.com/mytourbook/mytourbook/files/14175429/suunto.file.with.battery_level.zip) is containing these data 

@mff73 Can you provide a second fit file from another Suunto device. 

The problem is, that different devices can propably not be identified because the provided data in the fit file do not have a unique ID. 

![image](https://github.com/mytourbook/mytourbook/assets/1283445/01352361-ee01-4dca-be72-751ffd7c2f6f)


Compared with Garmin devices, each has a serial number which is also contained in the fit file and this is currently the ID to identify different devices

**Reply by wolfgang-ch** [reply]
The orange marked data are from your Suunto device, the green marked data are from my 5 Gamin/Shimano devices which all contain a unique serial number (blue border)

![image](https://github.com/mytourbook/mytourbook/assets/1283445/c61b3594-cb35-401a-a7f5-3723d0e89a45)

**Reply by mff73** [reply]
First, yes, i can provide such other files from other devices, let me find some.
Second, do i understand correctly : battery levels are/will be manage, not at the tour level, but at the/a device/sensor level ? meaning that you would need a unique serial/indentifier to affect such levels to this hardware ?
🤔

**Reply by wolfgang-ch** [reply]
>Second, do i understand correctly : battery levels are/will be manage, not at the tour level, but at the/a device/sensor level ? 

Yes, there is a sensor view https://mytourbook.sourceforge.io/mytourbook/index.php/documentation/show-tours/tour-sensor with all your sensors 

> meaning that you would need a unique serial/indentifier to affect such levels to this hardware ? 🤔

Yes, when a serial number is not available, then the manufacturer name/number plus product name/number will be used to identify a sensor.  These values are available in your fit files which you have sent to me.

When you have the same device twice (with same manufacturer and product), then it cannot be identified which sensor was used, in MT there will be only one.

**Reply by mff73** [reply]
Ok, thanks, it was not so clear to me :)
In my very own personnal case, i am not following two same "sensors" (watches) of the same manufacturer/model at the same time (meaning, no tour register the sames days). I may have/had same sensors but few months ago, so it will not be an issue, i think.
Next question :) If/When implemented for such suunto "sensors", could it be thinkable, once data will be imported for the sensor/watch (manuf name/product name), to manually create a duplicated sensor in MTB and then edit manually, for which one, battery values should be affected ? 
Like saying that, if one have multiple sensors of the same manuf/product, one could rename some of them into manuf1/product1, and then, if I understand well, MTB will import new fit battery values to """"official""""" sensor manuf/product, but MTB user could then reaffect, potentially, those values to manuf1/product1 ?

Pure theory for now, as i never ever seen any battery levels in my MTB for any of my devices, so i still don't see how to use it :)
And again, thanks to you and all developpers for all the job you are doing on this tools.

**Comment by wolfgang-ch** [comment]
> Next question :) If/When implemented for such suunto "sensors", could it be thinkable, once data will be imported for the sensor/watch (manuf name/product name), to manually create a duplicated sensor in MTB and then edit manually, for which one, battery values should be affected ?
> Like saying that, if one have multiple sensors of the same manuf/product, one could rename some of them into manuf1/product1, and then, if I understand well, MTB will import new fit battery values to """"official""""" sensor manuf/product, but MTB user could then reaffect, potentially, those values to manuf1/product1 ?

Do you really want to apply a device/sensor manually for each tour?

MT tries to automate this but it needs data which is/are stored in the fit file to identify a device uniquely

**Reply by mff73** [reply]
Hi Wolfang, 
I don't want to do it but if Suunto fit files are  not containing anything else that Brand/ProductName and Type, i am just imagining in advance what could be a workaround.
For each tour, i am applying manually tags, tour type, and doing my analysis on it, so i could definitively apply also a specific sensor 😊

**Reply by wolfgang-ch** [reply]
> I don't want to do it but if Suunto fit files are not containing anything else that Brand/ProductName and Type, i am just imagining in advance what could be a workaround.

I think, first you should test the next MT version (when available) then you can see what is still missing, see also screenshots in https://github.com/mytourbook/mytourbook/issues/1287#issuecomment-1938284698

> For each tour, i am applying manually tags, tour type, ...

Tour types can be automatically set when using the easy import feature, where the tour type depends on the average speed which mostly fits.

**Reply by mff73** [reply]
Oh, i didn't see your message, and the new feature 1287 :) Great ! Thanks a lot. Eager to test it.
👍
